File: ProductValidator.java
package org.jewelrystore;

import java.math.BigDecimal;
import java.util.regex.Pattern;

public class ProductValidator {
    private static final Pattern NAME_PATTERN = Pattern.compile("^[\\p{L}\\p{N}\\s.,'-]+$");
    private static final Pattern MATERIAL_PATTERN = Pattern.compile("^[\\p{L}\\s]+$");

    public static void validateProductId(int productId) {
        if (productId < 0) {
            throw new IllegalArgumentException("Product ID must be a non-negative integer.");
        }
    }

    public static void validateName(String name) {
        if (name == null || name.trim().isEmpty() || !NAME_PATTERN.matcher(name).matches()) {
            throw new IllegalArgumentException("Product name must contain only letters, numbers, spaces, and basic punctuation, and cannot be empty.");
        }
    }

    public static void validatePrice(BigDecimal price) {
        if (price == null || price.compareTo(BigDecimal.ZERO) <= 0) {
            throw new IllegalArgumentException("Price must be a positive number.");
        }
    }

    public static void validateStockQuantity(int stockQuantity) {
        if (stockQuantity < 0) {
            throw new IllegalArgumentException("Stock quantity must be a non-negative integer.");
        }
    }

    public static void validateMaterial(String material) {
        if (material == null || material.trim().isEmpty() || !MATERIAL_PATTERN.matcher(material).matches()) {
            throw new IllegalArgumentException("Material must contain only letters and spaces, and cannot be empty.");
        }
    }

    public static void validateDescription(String description) {
        if (description == null || description.trim().isEmpty()) {
            throw new IllegalArgumentException("Description cannot be empty.");
        }
        if (description.length() > 1000) {
            throw new IllegalArgumentException("Description must not exceed 1000 characters.");
        }
    }

    public static void validateProduct(Product product) {
        validateProductId(product.getProductId());
        validateName(product.getName());
        validatePrice(product.getPrice());
        validateStockQuantity(product.getStockQuantity());
        validateMaterial(product.getMaterial());
        validateDescription(product.getDescription());
    }
}

File: ProductValidator.py
from decimal import Decimal
import re
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .Product import Product


class ProductValidator:
    NAME_PATTERN = re.compile(r"^[\w\s.,'-]+$")
    MATERIAL_PATTERN = re.compile(r"^[A-Za-z\s]+$")

    @staticmethod
    def validate_product_id(product_id: int) -> None:
        if product_id < 0:
            raise ValueError("Product ID must be a non-negative integer.")

    @staticmethod
    def validate_name(name: str) -> None:
        if not name or not ProductValidator.NAME_PATTERN.match(name.strip()):
            raise ValueError(
                "Product name must contain only letters, numbers, spaces, and basic punctuation, and cannot be empty.")

    @staticmethod
    def validate_price(price: Decimal) -> None:
        if price is None or price <= Decimal('0'):
            raise ValueError("Price must be a positive number.")

    @staticmethod
    def validate_stock_quantity(stock_quantity: int) -> None:
        if stock_quantity < 0:
            raise ValueError("Stock quantity must be a non-negative integer.")

    @staticmethod
    def validate_material(material: str) -> None:
        if not material or not ProductValidator.MATERIAL_PATTERN.match(material.strip()):
            raise ValueError("Material must contain only letters and spaces, and cannot be empty.")

    @staticmethod
    def validate_description(description: str) -> None:
        if not description or not description.strip():
            raise ValueError("Description cannot be empty.")
        if len(description) > 1000:
            raise ValueError("Description must not exceed 1000 characters.")

    @staticmethod
    def validate_product(product: 'Product') -> None:
        ProductValidator.validate_product_id(product.get_product_id())
        ProductValidator.validate_name(product.get_name())
        ProductValidator.validate_price(product.get_price())
        ProductValidator.validate_stock_quantity(product.stock_quantity)
        ProductValidator.validate_material(product.material)
        ProductValidator.validate_description(product.description)